staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -292,6 +292,9 @@ func (c *threadSafeMap) deleteFromIndices(obj interface{}, key string) {
 			set := index[indexValue]
 			if set != nil {
 				set.Delete(key)
+				if len(set) == 0 {
+					delete(index, indexValue)
```

Comment:
it leaks when an empty namespace is deleted but the cache entry will retain. btw, i remember there's a duplicate pull months ago, but for now it seems leaked too.